wordstats/cognate_info.py

```diff
@@ -0,0 +1,324 @@
+import codecs
+import os
+from datetime import datetime
+
+from sqlalchemy import Table
+import configparser
+
+from wordstats.loading_from_hermit import load_language_from_hermit
+
+from .word_info import WordInfo, UnknownWordInfo
+from .utils.mem_footprint import total_size
+from .base_service import BaseService, Base
+from .config import MAX_WORDS
+from .metrics_computers import *
+from .cognate_files_path import *
+from .cognate_db import *
+from .getchunix import read_single_keypress
+from .edit_distance_function_factory import WordDistanceFactory
+
+
+class CognateInfo(object):
+    """
+
+        Responsibilities of this class:
+        - compute the candidates
+        - maintain the blacklist / whitelist (add_to_whitelist / add_to_blacklist / save_evaluation )
+        - load /save blacklist / whitelist / candidates from file / DB to memory
+
+
+
+    """
+
+    def __init__(self, primary, secondary, distance_computer_class: WordDistanceFactory):
+        """
+
+            either load from file, or compute if needed
+
+        :param primary:
+        :param secondary:
+        :param method_name:
+        """
+        self.primary = primary
+        self.secondary = secondary
+        self.whitelist = dict()
+        self.candidates = dict()
+        self.blacklist = dict()
+        self.distance_computer = distance_computer_class(primary, secondary)
+
+
+    def best_guess(self):
+        best_dict = dict(self.candidates)
+        for key in self.blacklist.keys():
+            best_dict.pop(key)
+
+        return best_dict
+
+    # generates candidates based on distance function func and word lists
+    # def apply_distance_metric(self, wordlist1, wordlist2, func):
+
+    def compute(self):
+        wordlist1 = list(load_language_from_hermit(self.primary).word_info_dict.keys())
+        wordlist2 = list(load_language_from_hermit(self.secondary).word_info_dict.keys())
+
+        self.candidates = dict()
+        for w1 in wordlist1:
+            for w2 in wordlist2:
+                if self.distance_computer.edit_distance(w1, w2):
+                    if w1 in self.candidates:
+                        self.candidates[w1].append(w2)
+                    else:
+                        self.candidates[w1] = [w2]
```

Comment:
there are two more `pythonic` ways of doing this: 

see: https://stackoverflow.com/a/17756005/1200070